sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_changelog_v1/query.sql

```diff
@@ -0,0 +1,265 @@
+    1 = COUNT(*) OVER (PARTITION BY subscription_id)
+),
+plans AS (
+  SELECT
+    id,
+    aggregate_usage,
+    amount,
+    billing_scheme,
+    created,
+    currency,
+    `interval`,
+    interval_count,
+    PARSE_JSON(metadata) AS metadata,
+    nickname,
+    product_id,
+    tiers_mode,
+    trial_period_days,
+    usage_type,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.plan_v1
+),
+subscriptions_history_with_plan_metadata AS (
+  SELECT
+    *,
+    JSON_VALUE(metadata.previous_plan_id) AS previous_plan_id,
+    LEAD(JSON_VALUE(metadata.previous_plan_id)) OVER (
+      PARTITION BY
+        subscription_id
+      ORDER BY
+        _fivetran_start
+    ) AS lead_previous_plan_id
+  FROM
+    subscriptions_history
+),
+subscriptions_history_with_end_plan_ids AS (
+  -- Determine the plan ID for the last record in each time span that a subscription had a particular plan.
+  SELECT
+    subscriptions_history.*,
+    CASE
+      -- The latest history record gets the current plan from its associated subscription item.
+      WHEN subscriptions_history._fivetran_active
+        THEN subscription_items.plan_id
+      -- A new `previous_plan_id` value means the previous record was the last record with that plan.
+      WHEN subscriptions_history.lead_previous_plan_id IS DISTINCT FROM subscriptions_history.previous_plan_id
+        THEN subscriptions_history.lead_previous_plan_id
+      ELSE NULL
+    END AS end_plan_id
+  FROM
+    subscriptions_history_with_plan_metadata AS subscriptions_history
+  JOIN
+    subscription_items
+  ON
+    subscriptions_history.subscription_id = subscription_items.subscription_id
+),
+subscriptions_history_with_plan_ids AS (
+  -- Fill in `plan_id` by getting the next `end_plan_id`.
+  SELECT
+    *,
+    FIRST_VALUE(end_plan_id IGNORE NULLS) OVER (
+      PARTITION BY
+        subscription_id
+      ORDER BY
+        _fivetran_start
+      ROWS BETWEEN
+        CURRENT ROW
+        AND UNBOUNDED FOLLOWING
+    ) AS plan_id
+  FROM
+    subscriptions_history_with_end_plan_ids
+),
+subscriptions_history_tax_rates AS (
+  SELECT
+    subscriptions_history.id AS subscription_history_id,
+    ARRAY_AGG(
+      STRUCT(
+        subscription_tax_rates.tax_rate_id AS id,
+        tax_rates.created,
+        tax_rates.description,
+        tax_rates.display_name,
+        tax_rates.inclusive,
+        tax_rates.jurisdiction,
+        PARSE_JSON(tax_rates.metadata) AS metadata,
+        tax_rates.percentage
+      )
+      ORDER BY
+        tax_rates.created
+    ) AS tax_rates
+  FROM
+    subscriptions_history
+  JOIN
+    `moz-fx-data-shared-prod`.stripe_external.subscription_tax_rate_v1 AS subscription_tax_rates
+  ON
+    subscriptions_history.subscription_id = subscription_tax_rates.subscription_id
+  JOIN
+    `moz-fx-data-shared-prod`.stripe_external.tax_rate_v1 AS tax_rates
+  ON
+    subscription_tax_rates.tax_rate_id = tax_rates.id
+    AND subscriptions_history._fivetran_end >= tax_rates.created
```

Comment:
Ideally we'd have some field to indicate when the tax rate was associated with the subscription, but I couldn't find such a field, so this is just a best-effort filter to avoid including tax rates with subscription history records from before the tax rate even existed.